app/src/main/java/de/tum/in/tumcampusapp/activities/FeedbackActivity.java

```diff
@@ -0,0 +1,363 @@
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import de.tum.in.tumcampusapp.R;
+import de.tum.in.tumcampusapp.activities.generic.BaseActivity;
+import de.tum.in.tumcampusapp.adapters.FeedbackThumbAdapter;
+import de.tum.in.tumcampusapp.auxiliary.Const;
+import de.tum.in.tumcampusapp.auxiliary.Utils;
+import de.tum.in.tumcampusapp.managers.LocationManager;
+
+public class FeedbackActivity extends BaseActivity {
+
+    private static final int REQUEST_TAKE_PHOTO = 11;
+    private static final int REQUEST_GALLERY = 12;
+    private static final int PERMISSION_LOCATION = 13;
+    private static final int PERMISSION_CAMERA = 14;
+    private static final int PERMISSION_FILES = 15;
+
+    private static final String EMAIL_PATTERN =
+            "^[_A-Za-z0-9-\\+]+(\\.[_A-Za-z0-9-]+)*@[A-Za-z0-9-]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
+    private Pattern pattern;
+
+    private CheckBox includeEmail, includeLocation;
+    private EditText feedbackView, customEmailView;
+
+    private String mCurrentPhotoPath;
+    private int feedbackTopic;
+    private String email;
+    private String lrzId;
+    private Location location;
+
+    private ArrayList<String> picPaths;
+
+    private RecyclerView.Adapter thumbAdapter;
+
+    public FeedbackActivity() {
+        super(R.layout.activity_feedback);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        feedbackTopic = 0; // General feedback by default
+
+        feedbackView = findViewById(R.id.feedback_message);
+        customEmailView = findViewById(R.id.feedback_custom_email);
+        includeEmail = findViewById(R.id.feedback_include_email);
+        includeLocation = findViewById(R.id.feedback_include_location);
+
+        lrzId = Utils.getSetting(this, Const.LRZ_ID, "");
+
+        initTopicSpinner(savedInstanceState);
+        initIncludeLocation(savedInstanceState);
+        initIncludeEmail(savedInstanceState);
+
+        if(savedInstanceState == null){
+            picPaths = new ArrayList<>();
+        } else {
+            picPaths = savedInstanceState.getStringArrayList(Const.FEEDBACK_PIC_PATHS);
+            feedbackView.setText(savedInstanceState.getString(Const.FEEDBACK_MESSAGE));
+        }
+
+        RecyclerView pictureList = findViewById(R.id.feedback_image_list);
+        pictureList.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, true));
+        thumbAdapter = new FeedbackThumbAdapter(picPaths);
+        pictureList.setAdapter(thumbAdapter);
+
+        pattern = Pattern.compile(EMAIL_PATTERN);
+    }
+
+    private void initIncludeLocation(Bundle savedInstanceState){
+        includeLocation.setOnClickListener(view -> {
+            if(includeLocation.isChecked() && location == null){
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
I'd suggest putting code that only works on specific versions in an extra method and annotate it with [`@RequiresApi`](https://developer.android.com/reference/android/support/annotation/RequiresApi.html)